clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java

```diff
@@ -183,8 +182,7 @@ public long brokerEpoch() {
 
     public Iterable<LeaderAndIsrPartitionState> partitionStates() {
         if (version() >= 2)
-            return () -> new FlattenedIterator<>(data.topicStates().iterator(),
-                topicState -> topicState.partitionStates().iterator());
+            return () -> data.topicStates().stream().flatMap(topicState -> topicState.partitionStates().stream()).iterator();
```

Comment:
@chia7712 Is it also equivalent performance wise?

Reply:
Indeed, we would want to verify that. It didn't look like it was when I originally checked, but I may have missed something.

Reply:
We should be generally careful about replacing simple code paths with more complex ones in Kafka. Kafka is a high performance system, so code readability is one factor, but not the only one.

Reply:
@dajac @ijuma Thanks for your response. You are right. the performance should be more important than code readability. will close this PR :)